cni: Update CNI version to v0.8.5

justaugustus opened the pull request:
Signed-off-by: Stephen Augustus <saugustus@vmware.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
- Update CNI version to v0.8.5
- Removes requirement for a Googler to upload new versions of CNI plugins by instead retrieving them from a new artifacts bucket (`gs://k8s-artifacts-cni/`), which has shared control between Release Engineering and SIG Network.
ref: https://github.com/kubernetes/k8s.io/pull/553

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
xref: Kubernetes CNI build/publish processes (https://github.com/kubernetes/sig-release/issues/245), Update CNI versions on k/release (https://github.com/kubernetes/release/pull/731)
ref: MIPS support (https://github.com/kubernetes/kubernetes/issues/87056)
Closes: https://github.com/kubernetes/kubernetes/issues/87228 (cc: @huchengze, @ydcool)

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Update CNI version to v0.8.5
```

justaugustus committed:
cni: Update CNI version to v0.8.5

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus commented:
/sig release network
/priority important-soon
/milestone v1.16

justaugustus commented:
/test pull-kubernetes-e2e-gce-device-plugin-gpu
/test pull-kubernetes-e2e-gce
/test pull-kubernetes-node-e2e-containerd
/test pull-kubernetes-node-e2e
/test pull-kubernetes-kubemark-e2e-gce-big
/test pull-kubernetes-e2e-gce-100-performance

justaugustus commented:
/retest

justaugustus commented:
Just as a note, I'm trying this out so I can better understand the process for packaging in general, as well as what needs to happen when we bump [external] dependencies.

An issue for documenting this process has already been opened here: https://github.com/kubernetes/sig-release/issues/245

So far, the experience for me has been:

- Look around the k/release repo and realize there are dependencies to potentially bump
- Open a PR to bump CNI in k/release: https://github.com/kubernetes/release/pull/731
- Get frustrated because this is not well documented
- Get some general details on what publishing behind the Google curtain looks like: https://github.com/kubernetes/sig-release/pull/657
- Search issues/PRs about CNI for hints on how to do this for k/k: https://github.com/kubernetes/kubernetes/pull/75455
- Open this PR
- See some tests fail

The following tests has consistently failed on this PR:
- pull-kubernetes-e2e-gce
- pull-kubernetes-e2e-gce-100-performance
- pull-kubernetes-e2e-gce-device-plugin-gpu
- pull-kubernetes-kubemark-e2e-gce-big
- pull-kubernetes-node-e2e
- pull-kubernetes-node-e2e-containerd

with the following message:
```
W0610 08:02:15.335] Analyzing: target //:push-build (3 packages loaded, 604 targets configured)
W0610 08:02:16.054] ERROR: /go/src/k8s.io/kubernetes/build/debs/BUILD:141:2: no such package '@kubernetes_cni_amd64//file': java.io.IOException: Error downloading [https://storage.googleapis.com/k8s-bazel-cache/https://storage.googleapis.com/kubernetes-release/network-plugins/cni-plugins-amd64-v0.8.1.tgz, https://storage.googleapis.com/kubernetes-release/network-plugins/cni-plugins-amd64-v0.8.1.tgz] to /bazel-scratch/.cache/bazel/_bazel_prow/48d5366022b4e3197674c8d6e2bee219/external/kubernetes_cni_amd64/file/kubernetes_cni.tgz: All mirrors are down: [GET returned 404 Not Found] and referenced by '//build/debs:kubernetes-cni-data-amd64'
W0610 08:02:16.074] ERROR: Analysis of target '//:push-build' failed; build aborted: Analysis failed
W0610 08:02:16.079] INFO: Elapsed time: 10.061s
W0610 08:02:16.079] INFO: 0 processes.
W0610 08:02:16.080] FAILED: Build did NOT complete successfully (3 packages loaded, 781 targets configured)
W0610 08:02:16.081] ERROR: Build failed. Not running target
```


I originally thought maybe this was a flake, but now I'm thinking that maybe the new `kubernetes-cni` package needs to be built and pushed before these will pass?


Adding a "few" folks here for insight and general visibility...
/cc @kubernetes/sig-network-misc @kubernetes/sig-network-test-failures @kubernetes/release-managers @sumitranr @aleksandra-malinowska @listx @thockin @philips @squeed @tpepper @calebamiles

justaugustus commented:
ref: CNI version sprawl throughout codebase (#75485)

BenTheElder commented:
was there a follow up to this somewhere else? we should bring this up in a sig channel / meeting or something ... this is a mess :/

justaugustus commented:
/retest

justaugustus commented:
@BenTheElder -- some of the convo is happening on https://github.com/kubernetes/release/pull/731 as well. I'll send something to SIG Network ML.

justaugustus commented:
Sent to ML: https://groups.google.com/d/topic/kubernetes-sig-release/PmIe5Er9qWU/discussion
(was also sent to SIG Network, but it might be caught in the moderator queue)

justaugustus committed:
cni: Update CNI download URLs to use new GCS bucket (k8s-artifacts-cni)

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus commented:
/retest

justaugustus commented:
/retest

justaugustus commented:
/retest

justaugustus commented:
/retest

BenTheElder commented:
[followed up with Stephen out of band, this builds now but e2e needs fixing]

BenTheElder commented:
build/dependencies.yaml also needs updating

BenTheElder commented:
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/78819/pull-kubernetes-e2e-gce/1154883242407497729/artifacts/e2e-fbceea84e5-674b9-master/kube-master-installation.log

> master configure.sh[790]: == Failed to download https://github.com/containernetworking/plugins/releases/download//cni-plugins-linux-amd64-v0.8.1.tgz. Retrying. ==

BenTheElder commented on cluster/gce/config-test.sh:
```diff
@@ -319,9 +319,9 @@ NODE_PROBLEM_DETECTOR_TAR_HASH="${NODE_PROBLEM_DETECTOR_TAR_HASH:-}"
-CNI_STORAGE_PATH="${CNI_STORAGE_PATH:-https://storage.googleapis.com/kubernetes-release/network-plugins}"
 CNI_VERSION="${CNI_VERSION:-}"
 CNI_SHA1="${CNI_SHA1:-}"
+CNI_STORAGE_PATH="${CNI_STORAGE_PATH:-https://github.com/containernetworking/plugins/releases/download/${CNI_VERSION}}"
```
If CNI_VERSION isn't explicitly set it won't be readable here. Instead this shouldn't include CNI_VERSION and we need to account for it below
```suggestion
CNI_STORAGE_PATH="${CNI_STORAGE_PATH:-https://github.com/containernetworking/plugins/releases/download/}"
```

BenTheElder commented on cluster/gce/gci/configure.sh:
```diff
@@ -236,10 +236,10 @@ function install-node-problem-detector {
   if [[ -n "${CNI_VERSION:-}" ]]; then
-      local -r cni_tar="cni-plugins-amd64-${CNI_VERSION}.tgz"
+      local -r cni_tar="cni-plugins-linux-amd64-${CNI_VERSION}.tgz"
       local -r cni_sha1="${CNI_SHA1}"
```
We should also compute "${CNI_STORAGE_PATH}/${CNI_VERSION}/${cni_tar}" to a local in each of these branches and then use that below.

justaugustus commented:
/test pull-kubernetes-e2e-gce
/test pull-kubernetes-kubemark-e2e-gce-big

justaugustus commented:
Good chatting with you, @BenTheElder!
Added your suggestions in https://github.com/kubernetes/kubernetes/pull/78819/commits/ccbb22c80151e5de61928bfdf23abac6256895a0 and https://github.com/kubernetes/kubernetes/pull/78819/commits/6528ea593f0d0b65e97723a1acf1aef1453ef3f4.

justaugustus commented:
/retest

justaugustus commented:
/area release-eng
/retest

justaugustus commented:
/retest

justaugustus commented:
/retest

aojea commented on cluster/gce/gci/configure.sh:
```diff
@@ -236,20 +236,24 @@ function install-node-problem-detector {
 
-  echo "Downloading cni binaries"
-  download-or-bust "${cni_sha1}" "${CNI_STORAGE_PATH}/${cni_tar}"
+  local -r cni_download_url="${CNI_GITHUB_DOWNLOAD_BASE_URL}/${cni_version}/${cni_tar}"
```
> CNI_GITHUB_DOWNLOAD_BASE_URL="${CNI_GITHUB_DOWNLOAD_BASE_URL:-https://github.com/containernetworking/plugins/releases/download/}"

should we remove the `/` before ${cni_version} to not have double slashes in the URL?

justaugustus replied:
Thanks for catching that, @aojea. Just fixed it!

squeed commented:
@justaugustus turns out there's also a bug in v0.8.1 - we'll be cutting v0.8.2 shortly.

BenTheElder commented:
this is pretty clearly failing across all of the e2es that pick it up... perhaps check if the config needs changes

aojea commented on build/dependencies.yaml:
```diff
@@ -46,7 +46,7 @@ dependencies:
 
   - name: "cni"
-    version: 0.7.5
+    version: 0.8.1
```
0.8.2

justaugustus replied:
Fixed. Thanks @aojea!

aojea commented:
/retest

xmudrii commented:
@justaugustus Is it planned to proceed with the CNI update this release?

justaugustus commented:
@xmudrii -- I won't have enough time to land this for 1.16, but I hope to early in the 1.17 cycle.
/milestone v1.17

josiahbjorgaard commented:
Bug triage for 1.17 here. Code freeze is Nov. 18th. Will this PR be merged by then? If not, can we move it out of the milestone?

josiahbjorgaard commented:
@justaugustus Are you still intending this for v.1.17?

saschagrunert commented:
v0.8.3 is out, this may be updated as well.

josiahbjorgaard commented:
/milestone v1.18

(Code freeze v1.17)

danwinship commented:
/assign @caseydavenport

justaugustus committed:
build: Don't attempt to use mirror for CNI plugin downloads

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus commented:
I've rebased and updated this PR to target CNI plugins v0.8.5.
In addition, some of the commits here have been broken out into https://github.com/kubernetes/kubernetes/pull/87562 to hopefully solve https://github.com/kubernetes/kubernetes/issues/87228 ahead of this merging.

justaugustus commented:
/retest

saschagrunert commented:
/retest

saschagrunert reviewed (approved):
/lgtm

aojea commented:
/retest

BenTheElder commented:
flakes:

[sig-storage] In-tree Volumes [Driver: gluster] [Testpattern: Pre-provisioned PV (default fs)] subPath should support non-existent path expand_more	2m49s
[sig-storage] Projected downwardAPI should provide node allocatable (memory) as default memory limit if the limit is not set [NodeConformance] [Conformance] expand_more	1m10s
[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] subPath should support existing directory expand_more

/retest

BenTheElder commented:
note that KIND CI has been on v0.8.5 roughly since it was released and running smoothly :-)

justaugustus commented:
/retest

justaugustus commented:
Flaking on `[sig-node] Downward API should provide default limits.cpu/memory from node allocatable [NodeConformance] [Conformance]`...
/retest

justaugustus commented:
More [flakes](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/78819/pull-kubernetes-e2e-gce/1221904193195872257/). This time:
- `[sig-apps] CronJob should delete successful finished jobs with limit of one successful job`
- `[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: dir-link] [Testpattern: Pre-provisioned PV (default fs)] subPath should support readOnly file specified in the volumeMount [LinuxOnly]`

/retest

justaugustus commented:
The saga [continues](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/78819/pull-kubernetes-e2e-gce/1221923319259860993/):
- `[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] subPath should support readOnly directory specified in the volumeMount`

/retest
cc: @kubernetes/ci-signal

aojea commented:
/retest
this time the job wasn't able to boot
> W0128 00:36:19.746] Server terminated abruptly (error code: 14, error message: '', log file: '/bazel-scratch/.cache/bazel/_bazel_prow/48d5366022b4e3197674c8d6e2bee219/server/jvm.out')

aojea commented:
/retest
```
[k8s.io] Pods should support pod readiness gates [NodeFeature:PodReadinessGate] expand_more	1m24s
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] volumes should allow exec of files on the volume expand_more	3m7s
[sig-storage] Downward API volume should provide container's memory limit [NodeConformance] [Conformance] expand_more	54s
[sig-storage] Projected downwardAPI should provide container's memory request [NodeConformance] [Conformance] expand_more	1m5s
[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: tmpfs] [Testpattern: Pre-provisioned PV (default fs)] subPath should support readOnly file specified in the volumeMount [LinuxOnly] expand_more
```

aojea commented:
/retest
```
[k8s.io] Pods should support pod readiness gates [NodeFeature:PodReadinessGate] expand_more	1m30s
[k8s.io] Container Runtime blackbox test on terminated container should report termination message [LinuxOnly] if TerminationMessagePath is set [NodeConformance] expand_more
```

justaugustus commented:
/hold for rebase on top of https://github.com/kubernetes/kubernetes/pull/87562

justaugustus commented:
/retest

justaugustus commented:
More flakes:
- `[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: dir] [Testpattern: Pre-provisioned PV (default fs)] subPath should support readOnly file specified in the volumeMount [LinuxOnly]`
- `[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: dir-link-bindmounted] [Testpattern: Pre-provisioned PV (default fs)] subPath should support readOnly directory specified in the volumeMount`

/retest

justaugustus commented:
Flake: `[sig-scheduling] PreemptionExecutionPath runs ReplicaSets to verify preemption running path`

/retest

justaugustus commented:
Appears to be stuck pending timeout in the push-build phase:
/test pull-kubernetes-e2e-gce-100-performance

justaugustus commented:
/retest

justaugustus commented:
[sig-api-machinery] Garbage collector should not be blocked by dependency circle [Conformance] expand_more	1m59s
[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] subPath should support file as subpath [LinuxOnly] expand_more

/retest

justaugustus commented:
Appears to be stuck in the "Copy Release" phase...
/test pull-kubernetes-e2e-gce

droslean commented:
/test pull-kubernetes-e2e-gce

dims commented:
/assign @thockin

thockin commented:
I can't imagine how this would be dangerous. </s>

I personally have low viz into CNI release process.  I am told there are "good" tests there, but I have not verified.  The KIND tests give some confidence, but really integration testing is about all we can do.

/lgtm
/approve

justaugustus commented:
Thanks Tim!
/hold cancel

cblecker commented:
/lgtm

tpepper commented:
I'm happy to see a bucket specific to CNI holding artifacts from which package building happens.  But unstated is to where the packages will be published.  We cannot and MUST NOT publish the updated packages to the existing package repositories, eg:

yliaog commented:
/hold

yliaog commented on cluster/gce/windows/k8s-node-setup.psm1:
```diff
@@ -953,18 +953,17 @@ function Configure-CniNetworking {
 #   SERVICE_CLUSTER_IP_RANGE
 function Configure_Dockerd_CniNetworking {
-  $CNI_RELEASE_VERSION = 'v0.8.2-gke.0'
+  $CNI_RELEASE_VERSION = 'v0.8.5'
```
this powershell module is used soley by GCE/GKE, please keep using the gs bucket (gs://gke-release/cni-plugins/v0.8.5-gke.0/) that our release team maintains.

justaugustus replied:
@yliaog -- Is there a reason we'd want the Windows tarballs to diverge from the ones provided by the CNI plugins maintainers & Kubernetes community?

The goal here is to provide everything from a single bucket, which has shared control between the CNI maintainers and Kubernetes Release Engineering.

Are there concerns on the Windows side that make this vary from the artifacts we're retrieving on Linux side (which also get used for the GCE/GKE tests)?

cc: @thockin

justaugustus replied:
(That is to say, I think all of these artifacts should come from the same place and I don't think this PR should be held.)

yliaog replied:
i'm going to create a PR to make them env variables

justaugustus replied:
@yliaog -- Sounds good. I'm going to unhold this then!
Would you mind tagging me on the PR when you make it?

/hold cancel

yliaog replied:
sure, i'll cc you the PR. thanks.

justaugustus commented:
@tpepper --

> I'm happy to see a bucket specific to CNI holding artifacts from which package building happens. 

To be clear, the only purpose for that bucket is to hold [CNI plugins release artifacts](https://github.com/containernetworking/plugins/releases), not any of what we republish in Kubernetes.

> But unstated is to where the packages will be published. We cannot and MUST NOT publish the updated packages to the existing package repositories, eg:
> 
> * https://packages.cloud.google.com/apt/dists/kubernetes-xenial/main/binary-amd64/Packages
> * https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64/
> 
> Doing so will break downstream users of the other Kubernetes packages.

The plan of record is discussed here:
- https://groups.google.com/d/topic/kubernetes-sig-release/yhf7hAqJEN0/discussion
- https://github.com/kubernetes/release/issues/885
- https://github.com/kubernetes/sig-release/issues/245

Basically:
- Publish no new versions of `kubernetes-cni` (the deb/rpm package) (we're currently already doing that as a side-effect of what you described)
- Move the CNI plugins into the `kubelet` package

Combined, that means:
- We're not introducing new dependencies to consider for older packages within the apt/yum repos
- Starting from whichever cycle we land this (targeting 1.18), the `kubernetes-cni` package would no dependency for the `kubelet` package

justaugustus commented:
/retest

justaugustus commented:
/retest

justaugustus commented:
/retest

justaugustus commented:
/retest

aojea commented:
:clap:
